synapseclient/client.py

```diff
@@ -621,6 +621,71 @@ def invalidateAPIKey(self):
         if self._is_logged_in():
             self.restDELETE("/secretKey", endpoint=self.authEndpoint)
 
+    @functools.lru_cache()
+    def get_user_profile_by_username(
+        self,
+        username: str = None,
+        sessionToken: str = None,
+    ) -> UserProfile:
+        """
+        Get the details about a Synapse user.
+        Retrieves information on the current user if 'id' is omitted or is empty string.
+        :param username:           The userName of a user
+        :param sessionToken: The session token to use to find the user profile
+        :returns: The user profile for the user of interest.
+
+        Example::
+            my_profile = syn.get_user_name_profile()
+            freds_profile = syn.get_user_name_profile('fredcommo')
+        """
+        if not isinstance(username, str) and username is not None:
+            raise TypeError("username must be string or None")
+        if isinstance(username, str):
+            principals = self._findPrincipals(username)
+            for principal in principals:
+                if principal.get("userName", None).lower() == username.lower():
+                    id = principal["ownerId"]
+                    break
+            else:
+                raise ValueError(f"Can't find user '{username}'")
+        else:
+            id = ""
+        uri = f"/userProfile/{id}"
+        return UserProfile(
+            **self.restGET(
+                uri, headers={"sessionToken": sessionToken} if sessionToken else None
+            )
+        )
+
+    @functools.lru_cache()
+    def get_user_profile_by_id(
+        self,
+        id: int = None,
+        sessionToken: str = None,
+    ) -> UserProfile:
+        """
+        Get the details about a Synapse user.
+        Retrieves information on the current user if 'id' is omitted.
+        :param id:           The ownerId of a user
+        :param sessionToken: The session token to use to find the user profile
+        :returns: The user profile for the user of interest.
+
+        Example::
+            my_profile = syn.getUserProfile()
```

Comment:
These examples need to be updated with the new function name

Reply:
updated in latest commit